src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpRequestNode.cs

```diff
@@ -85,11 +88,57 @@ public HttpBindingResult<HttpRequestMessage> TryGetHttpRequestMessage(HttpBindin
         }
         else
         {
-            success = false;
             diagnostics.AddRange(uriBindingResult.Diagnostics);
         }
 
-        if (success)
+        var headers =
+            HeadersNode?.HeaderNodes.Select(h => new KeyValuePair<string, string>(h.NameNode.Text, h.ValueNode.Text)).ToArray()
+            ??
+            Array.Empty<KeyValuePair<string, string>>();
+
+        foreach (var kvp in headers)
+        {
+            switch (kvp.Key.ToLowerInvariant())
+            {
+                case "content-type":
+                    if (request.Content is null)
+                    {
+                        request.Content = new StringContent("");
```

Comment:
Should this be handled globally in the `else` for the `if` block on line 136 below? Looks like Content being null is handled here - but if content is already an empty string we won't set `request.Content`.